doc/arch/adr-002-digest-mvp.md

```diff
@@ -0,0 +1,187 @@
+# Decision Record: MVP implementation of Digests
+
+## Introduction
+
+In January 2018 we started one of the latest major pieces of functionality that
+would be required to meet our MVP (Minimal viable product) and allow us to
+launch a Notify powered iteration of Email Alert API.
+
+This document serves to outline the architecture we have planned for this and
+explain the purposes of the individual components.
+
+## Specifics
+
+### Product decisions
+
+The need for a change in architecture was determined by product decisions which
+differed from the previous understanding of digest function explored in
+[adr-001](adr-001-notify-integration.md).
+
+The biggest decision that influenced the change was an intention to send each
+user a single digest email that contained all of their topics as is consistent
+with Govdelivery. We had previously intended to launch with the change that
+each subscription is a separate email.
+
+### Diagram
+
+![Sequence Diagram](../digests.png)
+
+### Components
+
+#### `DigestSchedulerService`
+
+There are two classes to represent the respective digest periods:
+`DailyDigestSchedulerService` and `WeeklyDigestSchedulerService`.
+
+Their responsibility is to start the process to create digests and to ensure
+that multiple digests are not started concurrently.
+
+##### Shall
+
+- Be interfaced by a scheduling service to run at a set time appropriate
+  for their period
+- Create a [`DigestRun`](#digestrun) model
+- Ensure that multiple digests for the same period are not run concurrently
+
+##### Should
+
+- Be configurable as to what times a digest runs from and until
+
+#### `DigestRun`
+
+A model which represents a distinct run of a period of digest (daily or
+weekly).
+
+Has a responsibility to persist data that is related to a full digest run for
+all subscribers.
+
+##### Shall
+
+- Store information such as
+  - Date digest is run on
+  - The period the digest is for
+  - The start time for the digest period
+  - The end time for the digest period
+
+##### Should
+
+- Store a timestamp for when all the emails were created for the digest, to act
+  as indicator it is complete.
+
+##### May
+
+- Have a unique index for digest date and period, however this would limit
```

Comment:
I imagined a re-run could re-use the record since I imagine the `completed_at` wouldn't be set if it was being re-run.

Reply:
There isn't currently a unique index on this. Nothing to stop us adding one obvs but just flagging that.